0000-catalog.md

```diff
@@ -0,0 +1,801 @@
+
+## Catalog_Product Actions
+
+### CatalogProductCreateAction
+
+The CatalogProductCreateAction adds a new catalog_product to state. The 
+catalog_product references a Grid Product for the shared item level master 
+data. The transaction should be submitted by an agent, which is identified by 
+its signing key, acting on behalf of the organization that corresponds to the
+owner in the create transaction. (Organizations and agents are defined by the 
+Pike smart contract.)
+
+```
+message CatalogProductCreateAction {
+    // catalog_id and product_id are used in deriving the state address
+    string catalog_id = 1
+    string product_id = 2 // Reference from Grid Product
+    // schema defined additional fields
+    repeated PropertyValues properties = 4;
+}
+```
+
+Validation requirements:
+
+- If a catalog_product with catalog_product_id and catalog_id already exists the
+  transaction is invalid.
+- If the Grid product the catalog_product is referencing does not exist in
+  state the transaction is invalid.
+- The signer of the transaction must be an agent in the Pike state and must
+  belong to an organization in Pike state, otherwise the transaction is invalid.
+- The agent must have the permission can_create_product for the organization,
+  otherwise the transaction is invalid.
+- If the product_namespace is GS1, the organization must contain a GS1 Company
+  Prefix in its metadata (gs1_company_prefixes), and the prefix must match the 
+  company prefix in the product_id, which is a GTIN (if GS1), otherwise the 
+  transaction is invalid.
+- The properties must be valid for the catalog_product schema; its properties
+  must only contain properties that are included in the catalog_product Schema.
+
+The inputs for CatalogProductCreateAction must include:
+
+- Address of the Agent submitting the transaction
+- Address of the Organization the Product is being created for
+- Address of the Catalog the catalog_product is related too
+- Address of the catalog_product namespace schema and the catalog_product’s
+  properties must match
+- Address of the Grid Product the catalog_product is referencing
+
+The outputs for CatalogProductCreateAction must include:
+
+- Address of the Catalog_Product created
+
+### CatalogProductUpdateAction
+
+CatalogProductUpdateAction updates an existing product in state. The
+transaction should be submitted by an agent, identified by its signing key, 
+acting on behalf of an organization that corresponds to the owner in the 
+product being updated. (Organizations and agents are defined by the Pike 
+smart contract.)
+
+```
+message CatalogProductUpdateAction {
+    // catalog_id and product_id are used in deriving the state address
+    string catalog_id = 1;
+    string product_id = 2;
+    // this will replace all properties currently defined
+    repeated PropertyValues properties = 4;
+}
+```
+
+Validation requirements:
+
+- If a catalog_product with catalog_product_id does not exist the transaction
```

Comment:
s/does not exist the/does not exist, the/

Reply:
✅